File: frontend/types/escalations.ts
export const escalationStatuses = [
  "open",
  "in_review",
  "approved",
  "rejected",
  "resolved",
] as const;

export const escalationSeverities = [
  "low",
  "medium",
  "high",
  "critical",
] as const;

export const escalationNoteTypes = ["system", "agent", "human"] as const;

export type EscalationStatus = (typeof escalationStatuses)[number];
export type EscalationSeverity = (typeof escalationSeverities)[number];
export type EscalationNoteType = (typeof escalationNoteTypes)[number];

export interface EscalationCase {
  case_id: string;
  conversation_id: string;
  trace_id: string | null;
  user_id: string;
  escalation_reason: string;
  severity: string;
  status: EscalationStatus;
  assigned_to: string | null;
  latest_agent: string | null;
  latest_summary: string | null;
  created_at: string;
  updated_at: string;
}

export interface EscalationCaseListResponse {
  cases: EscalationCase[];
  total: number;
}

export interface EscalationNote {
  note_id: string;
  case_id: string;
  author: string;
  note_type: EscalationNoteType;
  content: string;
  created_at: string;
}

export interface EscalationNoteListResponse {
  notes: EscalationNote[];
  total: number;
}

export interface EscalationAssignRequest {
  assigned_to: string;
  actor?: string;
  move_to_in_review?: boolean;
}

export interface EscalationStatusUpdateRequest {
  status: EscalationStatus;
  actor?: string;
}

export interface EscalationNoteCreateRequest {
  author: string;
  content: string;
  note_type: EscalationNoteType;
}

File: app/schemas/escalations.py
from pydantic import BaseModel, Field


class EscalationCaseResponse(BaseModel):
    case_id: str
    conversation_id: str
    trace_id: str | None = None
    user_id: str
    escalation_reason: str
    severity: str
    status: str
    assigned_to: str | None = None
    latest_agent: str | None = None
    latest_summary: str | None = None
    created_at: str
    updated_at: str


class EscalationNoteResponse(BaseModel):
    note_id: str
    case_id: str
    author: str
    note_type: str
    content: str
    created_at: str


class EscalationCaseListResponse(BaseModel):
    cases: list[EscalationCaseResponse]
    total: int


class EscalationNoteListResponse(BaseModel):
    notes: list[EscalationNoteResponse]
    total: int


class EscalationAssignRequest(BaseModel):
    assigned_to: str = Field(..., min_length=1, max_length=255)
    actor: str | None = Field(default=None, max_length=255)
    move_to_in_review: bool = True


class EscalationStatusUpdateRequest(BaseModel):
    status: str = Field(..., pattern="^(open|in_review|approved|rejected|resolved)$")
    actor: str | None = Field(default=None, max_length=255)


class EscalationNoteCreateRequest(BaseModel):
    author: str = Field(..., min_length=1, max_length=255)
    note_type: str = Field(..., pattern="^(system|agent|human)$")
    content: str = Field(..., min_length=1, max_length=4000)
